webapp/components/Modal/ReportModal.vue

```diff
@@ -65,52 +108,74 @@ export default {
       }, 1000)
     },
     async confirm() {
+      const { reasonCategory, reasonDescription } = this.form
       this.loading = true
-      try {
-        // TODO: Use the "modalData" structure introduced in "ConfirmModal" and refactor this here. Be aware that all the Jest tests have to be refactored as well !!!
-        // await this.modalData.buttons.confirm.callback()
-        await this.$apollo.mutate({
-          mutation: gql`
-            mutation($id: ID!) {
-              report(id: $id) {
-                id
-              }
-            }
-          `,
-          variables: { id: this.id },
+      // TODO: Use the "modalData" structure introduced in "ConfirmModal" and refactor this here. Be aware that all the Jest tests have to be refactored as well !!!
+      // await this.modalData.buttons.confirm.callback()
+      this.$apollo
+        .mutate({
+          mutation: reportMutation(),
+          variables: {
+            resourceId: this.id,
+            reasonCategory: reasonCategory.value,
+            reasonDescription,
+          },
         })
-        this.success = true
-        this.$toast.success(this.$t('report.success'))
-        setTimeout(() => {
-          this.isOpen = false
+        .then(({ _data }) => {
+          this.success = true
+          this.$toast.success(this.$t('report.success'))
           setTimeout(() => {
-            this.success = false
-            this.$emit('close')
-          }, 500)
-        }, 1500)
-      } catch (err) {
-        this.$emit('close')
-        this.success = false
-        switch (err.message) {
-          case 'GraphQL error: User':
-            this.$toast.error(this.$t('report.user.error'))
-            break
-          case 'GraphQL error: Post':
-            this.$toast.error(this.$t('report.contribution.error'))
-            break
-          case 'GraphQL error: Comment':
-            this.$toast.error(this.$t('report.comment.error'))
-            break
-        }
-      } finally {
-        this.loading = false
-      }
+            this.isOpen = false
+            setTimeout(() => {
+              this.success = false
+              this.$emit('close')
+            }, 500)
+          }, 1500)
+          this.loading = false
+        })
+        .catch(err => {
+          this.$emit('close')
+          this.success = false
+          switch (err.message) {
+            case 'GraphQL error: User':
+              this.$toast.error(this.$t('report.user.error'))
+              break
+            case 'GraphQL error: Post':
+              this.$toast.error(this.$t('report.contribution.error'))
+              break
+            case 'GraphQL error: Comment':
+              this.$toast.error(this.$t('report.comment.error'))
+              break
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-14T19:35:05Z" title="Monday, October 14th 2019, 9:35:05 pm +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -65,52 +132,73 @@ export default {
       }, 1000)
     },
     async confirm() {
+      const { reasonCategory, reasonDescription } = this.form
+
       this.loading = true
-      try {
-        // TODO: Use the "modalData" structure introduced in "ConfirmModal" and refactor this here. Be aware that all the Jest tests have to be refactored as well !!!
-        // await this.modalData.buttons.confirm.callback()
-        await this.$apollo.mutate({
-          mutation: gql`
-            mutation($id: ID!) {
-              report(id: $id) {
-                id
-              }
-            }
-          `,
-          variables: { id: this.id },
+      // TODO: Use the "modalData" structure introduced in "ConfirmModal" and refactor this here. Be aware that all the Jest tests have to be refactored as well !!!
+      // await this.modalData.buttons.confirm.callback()
+      this.$apollo
+        .mutate({
+          mutation: reportMutation(),
+          variables: {
+            resourceId: this.id,
+            reasonCategory: reasonCategory.value,
+            reasonDescription,
+          },
         })
-        this.success = true
-        this.$toast.success(this.$t('report.success'))
-        setTimeout(() => {
-          this.isOpen = false
+        .then(({ _data }) => {
+          this.success = true
+          this.$toast.success(this.$t('report.success'))
           setTimeout(() => {
-            this.success = false
-            this.$emit('close')
-          }, 500)
-        }, 1500)
-      } catch (err) {
-        this.$emit('close')
-        this.success = false
-        switch (err.message) {
-          case 'GraphQL error: User':
-            this.$toast.error(this.$t('report.user.error'))
-            break
-          case 'GraphQL error: Post':
-            this.$toast.error(this.$t('report.contribution.error'))
-            break
-          case 'GraphQL error: Comment':
-            this.$toast.error(this.$t('report.comment.error'))
-            break
-        }
-      } finally {
-        this.loading = false
-      }
+            this.isOpen = false
+            setTimeout(() => {
+              this.success = false
+              this.$emit('close')
+            }, 500)
+          }, 1500)
+          this.loading = false
+        })
+        .catch(err => {
+          this.$emit('close')
+          this.success = false
+          switch (err.message) {
+            case 'GraphQL error: User':
+              this.$toast.error(this.$t('report.user.error'))
+              break
+            case 'GraphQL error: Post':
+              this.$toast.error(this.$t('report.contribution.error'))
+              break
+            case 'GraphQL error: Comment':
+              this.$toast.error(this.$t('report.comment.error'))
+              break
```


What's the default/fallback here?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-15T18:04:06Z" title="Tuesday, October 15th 2019, 8:04:06 pm +02:00">Oct 15, 2019</time>_

---


the default now is to display the error from the `backend`, which can be that a user has already reported the resource